dimod/binary/binary_quadratic_model.py

```diff
@@ -624,6 +624,84 @@ def add_linear_equality_constraint(
                     u, v, 2 * lagrange_multiplier * ubias * vbias)
         self.offset += lagrange_multiplier * constant * constant
 
+    def add_linear_inequality_constraint(
+            self, terms: Iterable[Tuple[Variable, Bias]],
+            lagrange_multiplier: Bias, constant: Bias,
+            label: str,
+            slack_method: str = "log2",
+            slack_range: int = None,
+            discontinuous_slack: bool = False
+    ):
+        """Add a linear inequality constraint as a quadratic objective.
+
+        Adds a linear inequality constraint of the form
+        :math:`\\sum_{i} a_{i} x_{i} + C <= 0`
+        to the binary quadratic model as a quadratic objective. Only integral coefficient are handled correctly.
+        Try multiplying both sides of equations by factor of 10s to remove fractional coefficients.
+
+        Args:
+            terms (iterable/iterator):
+                An iterable of 2-tuples, (variable, bias).
+                Each tuple is evaluated to the term (bias * variable).
+                All terms in the list are summed.
+            lagrange_multiplier:
+                The coefficient or the penalty strength. This value is
+                multiplied by the entire constraint objective and added to the
+                bqm (it doesn't appear explicitly in the equation above).
+            constant:
+                The constant value of the constraint, C, in the equation above
+            label:
+                Label assigned to the new slack variables.
+            slack_method:
+                The method for adding slack. Only log2 method currently supported for BQM object. log2 adds
+                 log2(slack_range) number of binary variables to the constraint.
+            slack_range:
+                could be used to limit the range of slack variables.
+            discontinuous_slack:
+                If true it allows the lhs of the constraint be zero when slack_range is provided.
```

Comment:
If true, the lefthand side of the constraint can be zero when ``slack_range`` is provided.

Reply:
fixed